src/cve/utils/output_formatter.py

```diff
@@ -422,11 +413,20 @@ def _process_tool_output(content):
         Renderable markdown representation of the tool output.
     """
     if isinstance(content, str):
-        return f"<pre>{content}</pre>"
+        if content[0] == "[" and content[-1] == "]":
+            if content[1] != "{" and content[-2] != "}":
+                content = ast.literal_eval(content)
+            else:
+                content = {"source_documents": content}
+        else:
+            return f"<pre>{content}</pre>"
+    if isinstance(content, list):
+        list_md = '\n'.join(f'- {item}' for item in content) or []
+        return f"<pre>{list_md}</pre>"
```

Comment:
The current parsing is a bit fragile to unexpected tool output formats. Proposing this as an alternative:

```suggestion
    if isinstance(content, str):
        try:
            # Check if the string contains a list
            content = ast.literal_eval(content)
            # If not, raise an error to return the content as is
            if not isinstance(content, list):
                raise ValueError(f"Handling for tool output type {type(content)} is not implemented.")
        except Exception:
            return f"<pre>{content}</pre>"

    if isinstance(content, list):
        # Check if the list contains source documents
        if len(content) > 0 and isinstance(content[0], dict) and content[0].get("type", None) == "Document":
            content = {"source_documents": content}
        # If not, format and return the list as is
        else:
            list_md = '\n'.join(f'- {item}' for item in content) or []
            return f"<pre>{list_md}</pre>"

    # Format source documents
````

Reply:
this looks great, I have committed this as is, works locally for me